### ISSUE TITLE
[Python] Convert multidimensional arrays into FixedShapeTensorArrays automatically in `pyarrow.array`

### ISSUE THREAD

**Issue by alanhdu** [apache/arrow#35647]
### Describe the enhancement requested

Now that there is a canonical `fixed_shape_tensor` type, I was wondering whether there'd be any appetite for making `pyarrow.array` automatically convert numpy arrays with >1 dimension into `FixedShapeTensorArray`s.

You can do this manually by using `pa.FixedShapeTensorArray.from_numpy_ndarray`, but it seems more convenient for `pyarrow.array` to automatically support this.

If there's appetite for this, I think I'd be able to contribute a patch.

### Component(s)

Python

**Comment by AlenkaF** [comment]
I am personally interested in this idea 👍 
Could you elaborate a bit more how you plan to tackle this feature/which part of the codebase would you update, before making a PR?

**Comment by alanhdu** [comment]
Sure! I haven't fully groked the codebase, but I think there would be (potentially) two parts:

* I think the easiest thing would be to directly update https://github.com/apache/arrow/blob/3e4eaa917fa9b09a923d255adee520aa68a4e78c/python/pyarrow/array.pxi#L119 to call into `FixedShapeTensorArray.from_numpy_ndarray` when necessary. I think that basically means inserting another branch at https://github.com/apache/arrow/blob/3e4eaa917fa9b09a923d255adee520aa68a4e78c/python/pyarrow/array.pxi#L323 to check if `values.ndim > 1`.
* I could potentially update https://github.com/apache/arrow/blob/3e4eaa917fa9b09a923d255adee520aa68a4e78c/python/pyarrow/src/arrow/python/numpy_to_arrow.cc#L312 to return an ExtensionArray with the FixedShapeTensor if `ndim > 1`, but it's not clear to me how to go from the CExtensionArray to the Python type without modifying the Cython wrapper (and if we do thta, it's not clear there are advantages to not just doing it all in the Cython wrapper).

**Comment by AlenkaF** [comment]
> to return an ExtensionArray with the FixedShapeTensor if `ndim > 1`, but it's not clear to me how to go from the CExtensionArray to the Python type without modifying the Cython wrapper (and if we do thta, it's not clear there are advantages to not just doing it all in the Cython wrapper).

Yeah, I think if doing any kind of change in the array constructor that would fit in the Python C++ part of the codebase, because that is where all of the logic is implemented.

That said, I agree with @mariosasko that it would actually make sense to do it in a more generic way - to allow extension types the type inference when converting from numpy ndarrays for example. 

That said, the changes we are talking about here are not trivial.
@jorisvandenbossche what do you think?

**Comment by thisisnic** [comment]
This issue has been marked as stale because it has had no activity in the past 2 years. Please remove the stale label or comment below, or this issue will be closed in 14 days.
